packages/react-conformance/src/utils/getCallbackArguments.test.ts

```diff
@@ -0,0 +1,217 @@
+      throw new Error('Not supported');
+    },
+    writeFile: () => {
+      throw new Error('Not supported');
+    },
+
+    getSourceFile,
+    fileExists,
+  };
+}
+
+async function setupProgram(
+  rootNames: ts.CreateProgramOptions['rootNames'],
+  directoryJSON: DirectoryJSON,
+): Promise<ts.Program> {
+  vol.fromJSON(directoryJSON);
+
+  return ts.createProgram({
+    rootNames,
+    options: {},
+    host: createCompilerHost(),
+  });
+}
+
+describe('getCallbackArguments', () => {
+  describe('params', () => {
+    it('handles "null" as a param', async () => {
+      const program = await setupProgram(['Accordion.types.ts'], {
+        './Accordion.types.ts': 'export interface AccordionProps { onToggle: (e: null) => void; }',
+      });
+      const type = getCallbackArguments(program, 'Accordion.types.ts', 'AccordionProps', 'onToggle');
+
+      expect(type).toMatchObject({ e: null });
+    });
+
+    it('handles primitives as a param', async () => {
+      const program = await setupProgram(['Accordion.types.ts'], {
+        './Accordion.types.ts': `export interface AccordionProps {
+          onToggle: (a: string, b: number, c: boolean, d: undefined) => void;
+        }`,
+      });
+      const type = getCallbackArguments(program, 'Accordion.types.ts', 'AccordionProps', 'onToggle');
+
+      expect(type).toMatchObject({ a: 'String', b: 'Number', c: 'Boolean', d: undefined });
+    });
+
+    it('handles arrays', async () => {
+      const program = await setupProgram(['Accordion.types.ts'], {
+        './Accordion.types.ts': 'export interface AccordionProps { onToggle: (data: string[]) => void; }',
+      });
+      const type = getCallbackArguments(program, 'Accordion.types.ts', 'AccordionProps', 'onToggle');
+
+      expect(type).toMatchObject({ data: 'Array' });
+    });
+
+    it('handles simple type reference as a param', async () => {
+      const program = await setupProgram(['Accordion.types.ts'], {
+        './Accordion.types.ts': 'export interface AccordionProps { onToggle: (e: Event) => void; }',
+      });
+      const type = getCallbackArguments(program, 'Accordion.types.ts', 'AccordionProps', 'onToggle');
+
+      expect(type).toMatchObject({ e: 'Event' });
+    });
+
+    it('handles complex type reference as a param', async () => {
+      const program = await setupProgram(['Accordion.types.ts'], {
+        './Accordion.types.ts': `import * as React from 'react';
+      
+       export interface AccordionProps { onToggle: (e: React.MouseEvent) => void; }`,
+      });
+      const type = getCallbackArguments(program, 'Accordion.types.ts', 'AccordionProps', 'onToggle');
+
+      expect(type).toMatchObject({ e: 'React.MouseEvent' });
+    });
+
+    it('handles alias type as a param', async () => {
+      const program = await setupProgram(['Accordion.types.ts'], {
+        './Accordion.types.ts': `import * as React from 'react';
+      
+       type AccordionOnToggle = (e: React.MouseEvent) => void;
```

Comment:
is it realistic to test for a case where the alias is imported from another type file ?

Reply:
Added this test, it works 😀